### PR TITLE
added travis-ci configuration

### PR DESCRIPTION
**Pull request by frankmorgner**
@LudovicRousseau allowing travis-ci requires admin rights (which I don't currently have)

I plan to integrate coverity-scan with travis-ci.

### CODE REVIEW

**Comment by viktorTarasov** [comment]
@frankmorgner : Can you remind me why "--disable-sm" ?

**Comment by frankmorgner** [comment]
The dashes meant that all the three configurations are tested. Is there a reason *not* to test this option?

**Comment by viktorTarasov** [comment]
@frankmorgner : Ok, thanks. It was my misunderstanding.

**Comment by LudovicRousseau** [comment]
What do you need to allow travis-ci?

File: src/libopensc/iasecc-sdo.c [hunk 1]
```diff
@@ -1287,4 +1287,15 @@ iasecc_docp_copy(struct sc_context *ctx, struct iasecc_sdo_docp *in, struct iase
...
+
+/* we need to define the functions below to export them */
+#include "errors.h"
+
+int
+iasecc_sdo_encode_update_field()
+{
+	return SC_ERROR_NOT_SUPPORTED;
+}
+
```
Comment (LudovicRousseau) [inline src/libopensc/iasecc-sdo.c]: Is this change really needed to add travis-ci?

Maybe it should go in another independent patch?

**Comment by martinpaljak** [comment]
Reopening so that travis would see the PR and build it.

**Comment by viktorTarasov** [comment]
What for to scan PRs?
Seems that Coverity Scan keeps only the last scan result which should be reserved for <i>master</i> or for [dedicated branch](https://scan.coverity.com/travis_ci) .

**Comment by martinpaljak** [comment]
For coverity: yes, a dedicated branch is necessary. But for building pull requests (like jenkins) in different configurations, I see no reason why not to use Travis as well.

**Comment by viktorTarasov** [comment]
@martinpaljak : Ok, thanks, now I see.

**Comment by frankmorgner** [comment]
now verified with gcc, clang and cross compiling with mingw-w64

**Comment by frankmorgner** [comment]
@viktorTarasov coverity integration does not work with the configuration you pushed for some reason

File: src/libopensc/iasecc-sdo.c (hunk 1 above)
Comment (frankmorgner) [inline src/libopensc/iasecc-sdo.c]: Actually we should think about removing this function from the exports-file completely. @viktorTarasov do you really need it?

**Comment by viktorTarasov** [comment]
@frankmorgner
> coverity integration does not work with the configuration you pushed
'For me it works'. Travis and scan.
Where it stops to work for you? Access ?

@frankmorgner 
>  for some reason
consider it as private experimentation. Two years ago I tried to obtain public access to coverity-scan project dedicated to OpenSC but not succeeded. It was interesting to see how it could be done now.

**Comment by frankmorgner** [comment]
Simply reapply, I bet you get access now. So did I and I am already on the way of fixing dozens of problems.

Prepare your configuration to be replaced, I'll integrate it into mine. Coverity must then also be allowed inside the Travis CI account, which I think martin could do.

Am 27. Januar 2015 09:11:48 MEZ, schrieb viktorTarasov <notifications@github.com>:
>@frankmorgner
>> coverity integration does not work with the configuration you pushed
>'For me it works'. Travis and scan.
>Where it stops to work for you? Access ?
>
>@frankmorgner 
>>  for some reason
>consider it as private experimentation. Two years ago I tried to obtain
>public access to coverity-scan project dedicated to OpenSC but not
>succeeded. It was interesting to see how it could be done now. 
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenSC/OpenSC/pull/363#issuecomment-71606317

-- 
Frank Morgner

File: src/libopensc/iasecc-sdo.c (hunk 1 above)
Comment (viktorTarasov) [inline src/libopensc/iasecc-sdo.c]: @frankmorgner 
> do you really need it?

This function is used by 'local' version of SM module.

**Comment by frankmorgner** [comment]
Sorry for being unclear: look at the coverity overview. The scan results don't get updated despite your configuration.

Am 27. Januar 2015 09:11:48 MEZ, schrieb viktorTarasov <notifications@github.com>:
>@frankmorgner
>> coverity integration does not work with the configuration you pushed
>'For me it works'. Travis and scan.
>Where it stops to work for you? Access ?
>
>@frankmorgner 
>>  for some reason
>consider it as private experimentation. Two years ago I tried to obtain
>public access to coverity-scan project dedicated to OpenSC but not
>succeeded. It was interesting to see how it could be done now. 
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenSC/OpenSC/pull/363#issuecomment-71606317

-- 
Frank Morgner

**Comment by viktorTarasov** [comment]
From [coverity](https://scan.coverity.com/projects/4026?tab=overview):
> Jan 27, 2015
> Last Analyzed

Can you follow the links from [README.md](https://github.com/OpenSC/OpenSC/tree/coverity_scan) ?

**Comment by frankmorgner** [comment]
Sorry for the noise if that's the case. I'll verify later.

Am 27. Januar 2015 09:32:26 MEZ, schrieb viktorTarasov <notifications@github.com>:
>From [coverity](https://scan.coverity.com/projects/4026?tab=overview):
>> Jan 27, 2015
>> Last Analyzed
>
>Can you follow the links from
>[README.md](https://github.com/OpenSC/OpenSC/tree/coverity_scan) ?   
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenSC/OpenSC/pull/363#issuecomment-71608778

-- 
Frank Morgner

**Comment by martinpaljak** [comment]
On 27/01/15 10:25, Frank Morgner wrote:
> Prepare your configuration to be replaced, I'll integrate it into mine. Coverity must then also be allowed inside the Travis CI account, which I think martin could do.
I enabled Travis for OpenSC/OpenSC and the token in Viktor's changeset
seems to be legit.

m.

-- 
Martin
+372 5156 495

**Comment by viktorTarasov** [comment]
I've got an impression that
team members of GitHub project have automatic access to the travis-ci project created for this GitHub project. (Maybe I'm wrong and it was result of Martin's action.)
Do not know how if it can be accessed with non-team-member GitHub account. (Question to Martin?)

Team members also have automatic access to the summary of corresponding coverity-scan project.
To view defects they have to pass through request for access to the project admin. In case of  team member 'JenkinsForOpenSC' it was validated automatically.
Do not know how if it can be accessed with non-team-member GitHub account.

Links to test by the non-team-members:

Coverity Scan:
[Coverity Scan Build Status](https://scan.coverity.com/projects/4026)

Travis CI:
[Build Status](https://travis-ci.org/OpenSC/OpenSC)

**Comment by frankmorgner** [comment]
There is a concurrent project on coverity that does not get updated https://scan.coverity.com/projects/177. @LudovicRousseau, you may want to delete or disable it.

@viktorTarasov, you most likely want to have an exact copy of master in coverity_scan. My configuration works on both branches.

I also made the configuration compatible to osx builds as soon as they are not experimental anymore (see http://docs.travis-ci.com/user/multi-os/)

**Comment by LudovicRousseau** [comment]
@frankmorgner maybe the concurrent project is https://scan.coverity.com/projects/4026 :-)

https://scan.coverity.com/projects/177 has no automatic scan. So yes it could be removed.

**Comment by viktorTarasov** [comment]
@frankmorgner : Ok. The project is not very active, so probably we do not need coverity dedicated branch. Scan can be applied to the 'master', without PRs.

coverity-scan 'recommends':
> Up to   8 builds per week, with a maximum of 2 builds per day, for projects with 100K to 500K lines of code

@LudovicRousseau : The concurrent project was explicitly registered as GitHub project, using the new (for me) connector between coverity and GitHub. I suppose that it's more appropriated for situation. 

Do you know how to delete scan project ? I don't see button, no answer from scan-admin.

**Comment by LudovicRousseau** [comment]
No idea how to delete a coverity project. I guess the old coverity project will stay as is.

**Comment by viktorTarasov** [comment]
@LudovicRousseau : I was asking for my own, OpenSC dedicated coverity projects that I would like to delete.

**Comment by frankmorgner** [comment]
Coverity deleted the old project just now.

**Comment by martinpaljak** [comment]
FYI: I requested OSX support for OpenSC/OpenSC Travis project. Not useful in Coverity context but useful for build testing

**Pull request merged**